Directory.Build.targets

```diff
@@ -44,4 +44,62 @@
 
   <Import Project="eng\targets\Packaging.targets" Condition=" '$(MSBuildProjectExtension)' == '.csproj' " />
   <Import Project="eng\targets\ResolveReferences.targets" Condition=" '$(DisableReferenceRestrictions)' != 'true' AND '$(MSBuildProjectExtension)' == '.csproj' " />
+
+    <ItemGroup>
+    <PackageReference Include="xunit.runner.console" Version="2.4.1" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
+    <DistributedTestPlatform Include="Windows10.x64" />
+  </ItemGroup>
+
+  <Target Name="CreateTestPayload" Returns="@(TestPayload)">
+
+    <ItemGroup>
+      <_TargetFrameworks Remove="@(_TargetFrameworks)" />
+      <_TargetFrameworks Include="$(TargetFrameworks);$(TargetFramework)" />
+    </ItemGroup>
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+             Targets="_CreateTestPayloadInner"
+             Properties="TargetFramework=%(_TargetFrameworks.Identity)">
+      <Output TaskParameter="TargetOutputs" ItemName="TestPayload" />
+    </MSBuild>
+  </Target>
+
+  <Target Name="CollectXunitConsoleRunner" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PropertyGroup>
+      <XunitConsoleRunnerDir>$([System.IO.Path]::GetDirectoryName($(XunitConsole472Path)))</XunitConsoleRunnerDir>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <XunitConsoleRunnerFiles Include="$(XunitConsoleRunnerDir)/**/*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(XunitConsoleRunnerFiles)" DestinationFolder="$(PublishDir)" />
+  </Target>
+
+  <Target Name="_CreateTestPayloadInner" DependsOnTargets="Publish;CollectXunitConsoleRunner" Condition="'$(IsTestProject)' == 'true'" Returns="@(TestPayload)">
+
+    <ConvertToAbsolutePath Paths="$(PublishDir)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="PublishAbsoluteDir" />
+    </ConvertToAbsolutePath>
+
+    <ItemGroup>
+        <_CopyItems Include="$(MSBuildThisFileDirectory)test\vstest\runtests.cmd" Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'" />
+        <!-- switch to downloading this either before publish or on helix workers -->
+        <_CopyItems Include="$(MSBuildThisFileDirectory)test\vstest\dotnet-install.ps1" Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'" />
+        <_CopyItems Include="$(MSBuildThisFileDirectory)test\xunit\runtests.cmd" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(_CopyItems)" DestinationFolder="$(PublishAbsoluteDir)" />
+
+    <ItemGroup>
+      <TestPayload Include="$(PublishAbsoluteDir.TrimEnd('\').TrimEnd('\'))">
+        <TestAssembly>$(TargetFileName)</TestAssembly>
+        <TestName>$(MSBuildProjectName)/$(TargetFramework)</TestName>
+        <!-- <Command>runtests.cmd $(TargetFileName) $(_Microsoft_NETCore_App_Version)</Command> -->
+        <Command>runtests.cmd $(TargetFileName) 2.1.500</Command>
```

Comment:
I think you can use `$(NETCoreSdkVersion)` to get the .NET Core SDK version.

Reply:
Actually now I remember why I hardcoded this, the netcoreapp2.1 tests fail with the build sdk looking for 2.1.6:

```
The specified framework 'Microsoft.NETCore.App', version '2.1.6' was not found.
  - Check application dependencies and target a framework version installed at:
      C:\dotnetbuild\work\0374a9cf-3982-43d1-9cc8-33ea1a569f82\Payload\sdk\
  - Installing .NET Core prerequisites might help resolve this problem:
      http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
  - The .NET Core framework and SDK can be installed from:
      https://aka.ms/dotnet-download
  - The following versions are installed:
      2.1.5 at [C:\dotnetbuild\work\0374a9cf-3982-43d1-9cc8-33ea1a569f82\Payload\sdk\shared\Microsoft.NETCore.App]
```

Reply:
Oh. Try updating KoreBuild (run `build.cmd -u /t:Cow`). I updated it to 2.1.500 a couple weeks ago: https://github.com/aspnet/BuildTools/pull/875